### ISSUE TITLE
Discussion

### ISSUE THREAD

**Issue by rul124** [vertx-howtos/resilience4j-howto#1]
Hi team,

Thanks for this howto instruction, it's very helpful as I'm also working on the vert.x based retry mechanism. I have a question about why you are using this to generate promise:
```
    fun <T> executeFuture(
        retry: Retry,
        vertx: Vertx,
        supplier: Supplier<Future<T>>,
    ): Future<T> {
        val ctx = ContextInternal.current()
        val promise = if (ctx != null) ctx.promise() else Promise.promise<T>()
        AsyncRetryBlock(vertx, retry.asyncContext(), supplier, promise,).run()
        return promise.future()
    }
```
My guess is that you want this promise aware of current vertx context so any handlers attached to promise could be executed on the same context. But I wonder if it's a hard requirement? Will use promise created by `Promise.promise<T>()` break anything? Any suggestion? Thanks so much!

**Comment by Ladicek** [comment]
Yeah, I used a context-bound promise because it provides a _guarantee_ that the result will be emitted on the original context (which is what the caller generally expects). I'm not sure if that's strictly _necessary_, but if there's a way to guarantee correct behavior by construction, I'll take it as it's one less thing to think about.
